tools/telemetry-generator/src/handlers/appInsightsCustomDataTestHandler.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TelemetryClient } from "applicationinsights";
+
+/**
+ * This handler emits metrics to the Azure App Insights instance configured by the telemetryClient provided to this handler.
+ * This handler expects the 'telemetryClient' arg to be TelemetryClient class from the 'applicationinsights' Azure package.
+ */
+module.exports = function handler(fileData, telemetryClient: TelemetryClient): void {
+	console.log(`Found ${fileData.benchmarks.length} total benchmark tests to emit`);
+	for (const testData of fileData.benchmarks) {
+		for (const customDataKey of testData.customData) {
+			const customDataName = `${fileData.suiteName}_${testData.benchmarkName}_${customDataKey}`;
+
+			try {
+				console.log(
+					`emitting metric ${customDataName} with value ${testData.customData[customDataKey]}`,
+				);
+				telemetryClient.trackMetric({
+					name: customDataName,
+					value: testData.customData[customDataKey],
+					namespace: "performance_benchmark_customData",
+					properties: {
+						buildId: process.env.BUILD_ID,
+						branchName: process.env.BRANCH_NAME,
+						category: "performance",
+						eventName: "CustomBenchmark",
```

Comment:
I believe this is the correct `eventName` considering the [tag](https://github.com/microsoft/FluidFramework/blob/16494160f63266e9a89508f534f5d55195ad0d2b/tools/benchmark/src/mocha/customOutputRunner.ts#L40).